### DISCUSSION TITLE
使用api上传文件出现问题

### DISCUSSION THREAD

**Discussion by dplcz** [AlistGo/alist#1049] [Q&A]
![image](https://user-images.githubusercontent.com/73335978/167246303-b5a2d9ce-cdd4-4dd3-ae7e-57748dedadad.png)
返回结果为
{"code":403,"message":"Not allow upload","data":null}

是我请求有问题吗

**Comment by xhofe** [comment]
是的

**Reply by dplcz** [reply]
那正确的该是怎样呢

**Comment by Liner03** [comment]
兄弟你解决没

**Reply by dplcz** [reply]
用multipart/form-data上传文件必须要一个添加分割线才行，requests似乎没加这个功能

**Reply by dplcz** [reply]
接口还是 ./api/public/upload
from urllib3 import encode_multipart_formdata
from collections import OrderedDict
import requests

def upload_file(url, filename, author):
    headers['Authorization'] = author
    with open(filename, 'rb') as f:
        form_data = OrderedDict([('path', (None, '/夸克', 'multipart/form-data')),
                                 ('files', (f.name, f.read(), 'text/plain',))])
    form_data_sequence = encode_multipart_formdata(form_data, boundary='----WebKitFormBoundary9fvugkM1cAXO24AD')# 这个分割线可以改
    res = requests.post(url, data=form_data_sequence[0], headers=headers)
    print(res.content)

**Reply by Liner03** [reply]
请问路径是虚拟路径还是根文件夹路径，这两个我都试了都是{"code":500,"message":"path not found","data":null}

**Reply by dplcz** [reply]
![image](https://user-images.githubusercontent.com/73335978/179336875-306618a5-33c5-4bae-841b-88a3c1a42f2d.png)

**Reply by dplcz** [reply]
这个路径

**Reply by Liner03** [reply]
能留个联系方式吗 还是有些问题
